### ISSUE TITLE
IndexOutOfRangeException if LastChunk.Size on Archetype is 0

### ISSUE THREAD

**Issue by MindSwipe** [genaray/Arch#7]
When removing Entities from an Archetype either directly or via the World it sometimes throws an `IndexOutOfRangeException`, this is because for some reason the Archetype's `LastChunk.Size` is 0. An easy reproduction is the following:

Inside `EnumeratorTest.cs` add a new Test:

```csharp
[Test]
public void QueryRemove()
{
    var entities = new List<Entity>();
    world.GetEntities(description, entities);

    for (var i = 0; i < entities.Count; i++)
    {
        if (i == 5500)
            ;

        var entity = entities[i];
        world.Destroy(in entity);
    }
}
```

It will throw the exception when trying to remove the 5500th Entity, hence the if statement to set a breakpoint on the empty statement inside it

**Comment by genaray** [comment]
Thanks a lot ! :) Will look at this immediately and fix it.

**Comment by genaray** [comment]
> When removing Entities from an Archetype either directly or via the World it sometimes throws an `IndexOutOfRangeException`, this is because for some reason the Archetype's `LastChunk.Size` is 0. An easy reproduction is the following:
> 
> Inside `EnumeratorTest.cs` add a new Test:
> 
> ```cs
> [Test]
> public void QueryRemove()
> {
>     var entities = new List<Entity>();
>     world.GetEntities(description, entities);
> 
>     for (var i = 0; i < entities.Count; i++)
>     {
>         if (i == 5500)
>             ;
> 
>         var entity = entities[i];
>         world.Destroy(in entity);
>     }
> }
> ```
> 
> It will throw the exception when trying to remove the 5500th Entity, hence the if statement to set a breakpoint on the empty statement inside it

Issue fixed, the problem was that i forgot to "remove" the last chunk once it was empty... therefore `LastChunk` was not updated and the next removed entity caused such an exception.

Im gonna modify the tests a bit and than i'll publish the fix :)

**Comment by genaray** [comment]
Thanks again, great to see that someone actually uses or tests this project :) 

Adjusted some of the tests and added such a `DestroyAll` test. Works now correctly, fixed in [d0222f1](https://github.com/genaray/Arch/commit/d0222f15055a60726da549aeb986d1eee6c7cb83). The new nugget package is also already uploaded and will be available within a few minutes [Package](https://www.nuget.org/packages/Arch/1.0.10).

**Comment by MindSwipe** [comment]
Great! Thank you for the (very) fast response and fix.
